src/module/authentication.ts

```diff
@@ -83,6 +78,45 @@ class Authentication {
     // If the token is valid follow the next route.
     next();
   }
-}
 
-export default Authentication;
+  public static async grantDevAccess(req: Request, res: Response, next: NextFunction): Promise<void> {
+    const jwtToken: string = req.get('Authorization');
+
+    // Check if a authorization header is set.
+    if (jwtToken === undefined) {
+      new APIResponse(res, 401, {}, [
+        {
+          userMessage: 'Permission not granted.',
+          internalMessage: 'No JWT token was provided.',
+        },
+      ]).send();
+      return;
+    }
+
+    // If the header exists check the JWT token.
+    const validToken: JwtToken = await Authentication.verifyJwtToken(jwtToken);
+
+    try {
+      const userRepository = getRepository(User);
+      const user = await userRepository.findOneOrFail({
+        where: [{ login_name: validToken.username }],
+      });
+
+      if (validToken === null || !user.is_dev) {
```

Comment:
validToken.username in line 102 crashes if validToken === null.
The check for validToken === null needs to be done before the db access.